bpf-common/src/program.rs

```diff
@@ -357,10 +362,28 @@ impl Program {
                                     events.read
                                 );
                             }
-                            for buf in buffers.iter_mut().take(events.read) {
-                                let ptr = buf.as_ptr() as *const BpfEvent<T>;
-                                let event = unsafe { ptr.read_unaligned() };
-                                sender.send(Ok(event))
+                            for buffer in buffers.iter_mut().take(events.read) {
+                                // dbg!(buffer.len());
+                                // dbg!(event_size);
+                                let mut buffer =
+                                    std::mem::replace(buffer, BytesMut::with_capacity(buffer_size));
+                                let ptr = buffer.as_ptr() as *const RawBpfEvent<T>;
+                                let raw = unsafe { ptr.read_unaligned() };
+                                buffer.advance(event_size);
+                                // dbg!(buffer.len());
+                                // dbg!(raw.buffer.buffer_len);
```

Comment:
I think I wanted to keep those around in case I had bugs and needed to debug them. It should be tested enough now and I could always add them back later. Removing them.